Feature/kanban board pt2 tasks

semensid2020 opened the pull request.

semensid2020 committed:
js_routes generated, @material-ui packages added

semensid2020 committed:
fetching tasks from API

semensid2020 committed:
tasks display on board added

semensid2020 committed:
'load more tasks' button and display of tasks number

semensid2020 committed:
drag-n-drop task state change added

semensid2020 committed:
button 'new task' added

semensid2020 committed:
Adding new task from kanban implemented

semensid2020 committed:
Editing and destroying tasks from kanban implemented

YJorge commented on node_modules/.yarn-integrity:
```diff
@@ -11,10 +11,13 @@
     "@babel/plugin-proposal-private-methods@^7.18.6",
```
этого в пр быть не должно

semensid2020 replied:
@YJorge 
Чего именно быть не должно? в целом node_modules? или пакета конкретного

YJorge replied:
в целом node_modules. Модули все равно будут у каждого разработчика скачиваться отдельно

YJorge commented on app/javascript/utils/fetchHelper.js:
```diff
@@ -0,0 +1,66 @@
+    return axios.put(url, body).then(camelize);
+  },
+
+  delete(url, json) {
```
для delete body не нужен

YJorge replied:
json надо тоже убрать, если для метода он не нужен

YJorge replied:
@semensid2020

YJorge commented on app/javascript/repositories/TasksRepository.js:
```diff
@@ -0,0 +1,29 @@
+    return FetchHelper.post(path, task);
+  },
+
+  destroy(id) {
```
собственно тут боди и не передается

YJorge commented:
запусти, пжлст, еще раз деплой в уфиззи, а то сборка заэкспайрилась

semensid2020 commented:
@YJorge
Пересоздал среду, 

по правкам указанным выше пока не вносил изменения

YJorge commented:
1. Нельзя залогиниться в акк

https://github.com/semensid2020/task-manager1/assets/81574649/322d9c10-94bb-47a4-89cd-daf2112dee90

2. Надо поправить стили колонок, чтобы были фиксированного размера

https://github.com/semensid2020/task-manager1/assets/81574649/1c5be4a9-f7f4-4e09-a8d4-13f193faf38f

semensid2020 committed:
node_modules folder deleted from repo (after PR#19 review)

semensid2020 committed:
various fixes (after PR#19 review)
1. Unnecessary second argument removed from delete requests
2. Column width error fixed for empty columns

Additonally:
3. Environment is displayed in header (just for convinience)

semensid2020 committed:
trying to catch problems with db:seed at uffizzi

semensid2020 committed:
fixed seeds: incorrect email format was used

semensid2020 commented:
@YJorge 
видео одинаковые загрузились :(
Не удаётся войти в акк админа по email'у   admin@localhost?

Выяснил, это и не должно было работать. Такой акк взял в сидах из туториала, валидации какие-то немного другие мы согласовали (ещё в PR#6). Их этот email не проходит. Прошу использовать вот такую учетку: 
логин:   admin@mail.gen 
пароль: admin

Другие замечания тоже исправил.

YJorge commented:
> @YJorge видео одинаковые загрузились :( Не удаётся войти в акк админа по email'у admin@localhost?
> 
> Выяснил, это и не должно было работать. Такой акк взял в сидах из туториала, валидации какие-то немного другие мы согласовали (ещё в PR#6). Их этот email не проходит. Прошу использовать вот такую учетку: логин: [admin@mail.gen](mailto:admin@mail.gen) пароль: admin
> 
> Другие замечания тоже исправил.

попробуй создать акк с мылом 1@1(admin@localhost тоже имеет такой паттерн). Создать такую учетку можно, но залогиниться в неё нельзя

YJorge commented on db/seeds.rb:
```diff
@@ -1,12 +1,14 @@
+admin = Admin.find_or_create_by(first_name: 'admin', last_name: 'admin', email: 'admin@mail.gen')
 admin.password = 'admin'
-admin.save
+p admin if admin.save
```
не нужно выводить в stdout какие либо данные

YJorge commented on node_modules/.yarn-integrity:
```diff
@@ -1,1586 +0,0 @@
-{
```
откуда эти пакеты?  они в девелопе вообще не должны быть

semensid2020 replied:
@YJorge 
Какие именно?

YJorge replied:
любые node_modules

semensid2020 replied:
Ну теперо не будет - я их удалил из репозитория в коммите ( 6561441 ).
Как попали в репозиторий - не знаю, если честно затрудняюсь ответить...
Папка вроде как в git_ignore, но шото не сработал он.

YJorge replied:
У тебя в ide они отображаются как игнорируемые?
<img width="308" alt="image" src="https://github.com/semensid2020/task-manager1/assets/81574649/2275f676-f803-4bb2-afe0-35886674008b">

semensid2020 replied:
@YJorge 
серым залиты, да
![image](https://github.com/semensid2020/task-manager1/assets/46713486/fc07ca97-a5ff-46a4-82f3-bac93ca70369)

semensid2020 replied:
@YJorge 
Георгий, (я правильно обращаюсь?)
я привел в соответствие валидацию email в модели user.rb  и на форме   /forms/session_form.rb
Взял тот паттерн для валидации, который даётся в [туториале ](https://learn.dualboot.ru/courses/5/lessons/76/practices/1144)
дефолтная учетка из сидов admin@localhost тут не пройдет, прошу использовать:  admin@mail.gen    +  пасс admin

Также убрал вывод лишней инфы в stdout.      Среда пересоздалась.

YJorge replied:
Да, Георгий - правильно. 

Нельзя создать нового пользователя из админки

https://github.com/semensid2020/task-manager1/assets/81574649/6eba7142-7358-4908-8c47-db4c5d024300

semensid2020 replied:
@YJorge 
доброе утро, исправил. 
не хочется пенять на туториал, но `render(:new_admin_user_url)` тоже было взято из [него](https://learn.dualboot.ru/courses/5/lessons/78/practices/1175) 😄 
Вот то что не допроверил верно ли его использовать - мой косяк, да.

Имеет смысл собрать в кучу эти неточности, будет кто-то смотреть их? чтобы в туториал внести правки

p.s.
Также прошелся ещё раз по сайту: закрыл рендеринг страниц  "Sign in" и "Sign up" если юзер залогинен, 
поправил на странице "Admin page" сортировку  записей по нажатию на название колонки - id, first_name, last_name...  Вроде не вижу недоработок больше.

semensid2020 replied:
@YJorge 
доброе утро.
обновил среду (а то она неудачно "просыпается" всегда).
Если есть возможность - буду рад получить approve, чтобы двигаться дальше.

p.s. напомню, что админская учётка:
логин: admin@mail.gen
пароль: admin

YJorge replied:
@semensid2020 не всегда есть возможность оперативно проверять ПР. Если отправил на проверку, его точно рано или поздно кто-нибудь проверит

YJorge replied:
По поводу ошибок в туториале - это все таки не туториал, а курс по обучению. Где-то ошибки добавлены специально, что бы у учеников не все шло как по-маслу. При реальной разработке ошибки тоже не планируются, но они есть. Это ответственность разработчика проверить, что его код работает, а если нет, то исправить

semensid2020 replied:
@YJorge 
Спасибо. Хорошего дня =)

YJorge replied:
Взаимно

semensid2020 committed:
email format validation and stdout fixed (after PR#19 review)

semensid2020 committed:
some more fixes (after PR#19 review)
1. Fixed rendering of user creation form in case ELSE
2. Removed unnecessary second argument

semensid2020 committed:
additional fixes (during PR#19 review)
1. ransack sorting by attributes (id, first_name...) fixed
2. sign_up page is not rendered if user already signed_in
Additionally:
3. current commit hash is displayed in header (for convinience)

semensid2020 committed:
Unnecessary info removed from header

semensid2020 committed:
additional fixes 2 (during PR#19 review)
1. sign_in page is not rendered if user already signed_in
